From b00b033f14089b79af277186e72d75e7655e3d3a Mon Sep 17 00:00:00 2001
From: Li Jun <b47624@freescale.com>
Date: Fri, 9 Jan 2015 14:37:24 +0800
Subject: [PATCH 0058/5242] MLK-10085-4 usb: doc: chipidea: imx: add
 imx6-usb-charger-detection property

commit  b0e156c9c2ae401986b0fe1feb425440bbacdf1a from
https://source.codeaurora.org/external/imx/linux-imx.git

It is used to indicate whether we use SoC's usb charger
detection or not. Besides, we add anatop phandle since
we need to use anatop register to do most of charger detect operations.

Signed-off-by: Peter Chen <peter.chen@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../devicetree/bindings/usb/ci-hdrc-usb2.txt       |   22 +++++++++++++++++++-
 1 file changed, 21 insertions(+), 1 deletion(-)

diff --git a/Documentation/devicetree/bindings/usb/ci-hdrc-usb2.txt b/Documentation/devicetree/bindings/usb/ci-hdrc-usb2.txt
index 2e93181..9cb0098 100644
--- a/Documentation/devicetree/bindings/usb/ci-hdrc-usb2.txt
+++ b/Documentation/devicetree/bindings/usb/ci-hdrc-usb2.txt
@@ -88,8 +88,14 @@ i.mx specific properties
 - over-current-active-high: over current signal polarity is high active,
   typically over current signal polarity is low active.
 - external-vbus-divider: enables off-chip resistor divider for Vbus
+- imx6-usb-charger-detection: enable imx6 usb charger detect function,
+  only set it when the user wants SoC usb charger detection capabilities.
+  If the user wants to use charger IC's usb charger detection capabilities,
+  please do not set it.
+- fsl,anatop: phandle for anatop module, anatop module is only existed
+  at imx6 SoC series.
 
-Example:
+Examples:
 
 	usb@f7ed0000 {
 		compatible = "chipidea,usb2";
@@ -109,3 +115,17 @@ Example:
 		mux-controls = <&usb_switch>;
 		mux-control-names = "usb_switch";
 	};
+
+	usb@02184000 { /* USB OTG */
+		compatible = "fsl,imx6q-usb", "fsl,imx27-usb";
+		reg = <0x02184000 0x200>;
+		interrupts = <0 43 0x04>;
+		fsl,usbphy = <&usbphy1>;
+		fsl,usbmisc = <&usbmisc 0>;
+		disable-over-current;
+		external-vbus-divider;
+		imx6-usb-charger-detection;
+		fsl,anatop = <&anatop>;
+		maximum-speed = "full-speed";
+		tpl-support;
+	};
-- 
1.7.9.5

